specs/merge/beacon-chain.md

```diff
@@ -239,6 +260,54 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
 
 ### Execution payload processing
 
+#### `is_valid_gas_limit`
+
+```python
+def is_valid_gas_limit(payload: ExecutionPayload, parent: ExecutionPayloadHeader) -> bool:
+    parent_gas_limit = parent.gas_limit
+
+    # Check if the payload used too much gas
+    if payload.gas_used > payload.gas_limit:
+        return False
+
+    # Check if the payload changed the gas limit too much
+    if payload.gas_limit >= parent_gas_limit + parent_gas_limit // GAS_LIMIT_DENOMINATOR:
+        return False
+    if payload.gas_limit <= parent_gas_limit - parent_gas_limit // GAS_LIMIT_DENOMINATOR:
+        return False
+
+    # Check if the gas limit is at least the minimum gas limit
+    if payload.gas_limit < MIN_GAS_LIMIT:
+        return False
+
+    return True
+```
+
+#### `compute_base_fee_per_gas`
+
+```python
+def compute_base_fee_per_gas(payload: ExecutionPayload, parent: ExecutionPayloadHeader) -> uint64:
```

Comment:
There was a prior discussion about using slot and counting empty slot as an empty block wrt base fee and capacity. We need to revise that conversation before finalizing this formula

there were pros and cons each way and definitely wasnt a consensus on the right way to do it